lib/multi.c

```diff
@@ -3937,3 +3939,24 @@ static void multi_xfer_bufs_free(struct Curl_multi *multi)
   multi->xfer_ulbuf_len = 0;
   multi->xfer_ulbuf_borrowed = FALSE;
 }
+
+struct Curl_easy *Curl_multi_get_handle(struct Curl_multi *multi,
+                                        curl_off_t id)
+{
+  struct Curl_easy *data = multi->easyp;
+  struct Curl_llist_element *e;
+
+  if(id >= 0) {
+    for(data = multi->easyp; data; data = data->next) {
+      if(data->id == id)
+        return data;
+    }
+    /* may be in msgsent queue? */
+    for(e = multi->msgsent.head; e; e = e->next) {
```

Comment:
It might also be in the `&multi->pending` list. See `MSTATE_CONNECT` handling.

Reply:
Added.